pkg/operator/ceph/cluster/mon/spec.go

```diff
@@ -134,7 +134,55 @@ func (c *Cluster) makeDeploymentPVC(m *monConfig) (*v1.PersistentVolumeClaim, er
  * Pod spec
  */
 
-func (c *Cluster) makeMonPod(monConfig *monConfig, hostname string) *v1.Pod {
+func (c *Cluster) setPodAffinity(pod *v1.PodSpec) {
+	pod.NodeSelector = nil
+
+	// try to avoid scheduling monitors within the same available zone
+	//
+	// TODO: the topology key should likely be exposed through the CRD. However,
+	// this does preserve the policy of the current explicit scheduler in Rook
+	// which always looks for the `zone` topology key.
+	pod.Affinity = &v1.Affinity{
```

Comment:
If the user specified affinity or antiaffinity in the cluster CR, will it still be applied here? Or we are just overriding it?

Reply:
I've updated this in https://github.com/rook/rook/pull/3668/commits/a49f33ce80a799caaa312c3725158654dfa848cd basically we expect the topology key to be specified in the CRD, simply because we can't know which topology key will be used in a particular setup.

the topology key in the CRD can go in a `Preferred*` or `Required*` affinity setting, then in the updated code we do not override this, but merge in the only hard coded affinity settings which are the mon-pod anti affinity that controls colocation on nodes.